src/MetadataUtility/Audio/Wave.cs

```diff
@@ -334,6 +339,12 @@ public static Fin<RangeHelper.Range> ScanForChunk(Stream stream, RangeHelper.Ran
                 // advance our offset counter by the 8 bytes we just read
                 offset += read;
 
+                // check the chunk length falls within the bounds of the file
+                if (offset + length > stream.Length)
+                {
+                    return InvalidChunk;
+                }
+
```

Comment:
Added this to make sure the sub chunk length doesn't exceed the bound of the file. Now the data chunk will throw a Fin and format chunk will process.